pkg/phlaredb/block_querier.go

```diff
@@ -767,6 +769,37 @@ func MergeProfilesPprof(ctx context.Context, stream *connect.BidiStream[ingestv1
 	return nil
 }
 
+func Series(ctx context.Context, req *ingestv1.SeriesRequest, blockGetter BlockGetter) (*ingestv1.SeriesResponse, error) {
+	queriers, err := blockGetter(ctx, model.TimeFromUnix(req.Start), model.TimeFromUnix(req.End))
+	if err != nil {
+		return nil, err
+	}
+
+	var labelsSet []*typesv1.Labels
+	var lock sync.Mutex
+	group, ctx := errgroup.WithContext(ctx)
+
+	for _, q := range queriers {
+		group.Go(util.RecoverPanic(func() error {
+			labels, err := q.Series(ctx, req)
+			if err != nil {
+				return err
+			}
+
+			lock.Lock()
+			labelsSet = append(labelsSet, labels...)
+			lock.Unlock()
+			return nil
+		}))
+	}
+	err = group.Wait()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ingestv1.SeriesResponse{LabelsSet: labelsSet}, nil
```

Comment:
This will contain many duplicates, I would sugest at least here we should order and deduplicate. Maybe even at every step.

Reply:
Fixed in https://github.com/grafana/pyroscope/pull/2431/commits/a9dc85efb2565b51c8affb9b825b4c55dcfe0e66